static/app/components/smartSearchBar/index.tsx

```diff
@@ -749,13 +769,222 @@ class SmartSearchBar extends React.Component<Props, State> {
+        type: 'tag-operator' as ItemType,
+        value: ':<=',
+        desc: t(`${tag}:<=[value] is less than or equal to`),
+      },
+      [TermOperator.GreaterThan]: {
+        type: 'tag-operator' as ItemType,
+        value: ':>',
+        desc: t(`${tag}:>[value] is greater than`),
+      },
+      [TermOperator.LessThan]: {
+        type: 'tag-operator' as ItemType,
+        value: ':<',
+        desc: t(`${tag}:<[value] is less than`),
+      },
+      [TermOperator.Equal]: {
+        type: 'tag-operator' as ItemType,
+        value: ':=',
+        desc: t(`${tag}:=[value] is equal to`),
+      },
+      [TermOperator.NotEqual]: {
+        type: 'tag-operator' as ItemType,
+        value: '!:',
+        desc: t(`!${tag}:[value] is not equal to`),
+      },
+    };
+    return validOps.map(op => operatorMap[op]);
+  }
+
+  getValidOps(filterToken: TokenResult<Token.Filter>): readonly TermOperator[] {
+    const types = filterToken.invalid?.expectedType ?? [filterToken.filter];
+    const filterType = interchangeableFilterOperators[types[0]]
+      ? interchangeableFilterOperators[types[0]][0]
+      : types[0];
+    const config = filterTypeConfig[filterType];
+    return config.validOps ?? [''];
+  }
+
+  generateOpAutocompleteGroup(
+    validOps: readonly TermOperator[],
+    tagName: string
+  ): AutocompleteGroup {
+    const operatorItems = this.generateOpAutocompleteEntries(validOps, tagName);
+    return {
+      searchItems: operatorItems,
+      recentSearchItems: undefined,
+      tagName: '',
+      type: 'tag-operator' as ItemType,
+    };
+  }
+
+  withinTokenLocation(node: {location: LocationRange}, position: number) {
+    return position >= node.location.start.offset && position <= node.location.end.offset;
+  }
+
+  getCursorToken = (AST: ParseResult, cursor: number): ParseResult[number] | null => {
+    // traverse AST to find first matching filter based on cursor position
+    for (const node of AST) {
+      if (node !== Token.Spaces && this.withinTokenLocation(node, cursor)) {
+        // traverse into a logic group to find specific filter
+        if (node.type === Token.LogicGroup) {
+          return this.getCursorToken(node.inner, cursor);
+        }
+        return node;
+      }
+    }
+    return null;
+  };
+
   updateAutoCompleteItems = async () => {
     if (this.blurTimeout) {
       clearTimeout(this.blurTimeout);
       this.blurTimeout = undefined;
     }
 
     const cursor = this.getCursorPosition();
+    const {organization} = this.props;
+    const AST = this.state.parsedQuery;
+    if (AST && organization.features.includes('search-syntax-highlight')) {
+      const filterNode = this.getCursorToken(AST, cursor);
+      if (filterNode && filterNode.type === Token.Filter) {
+        // check if we are on the tag, value, or operator
+        if (this.withinTokenLocation(filterNode.value, cursor)) {
+          const node = filterNode.value;
+          const tagName = filterNode.key.text;
```

Comment:
There's a helper utility for this, but tag names won't actually always be just `filterNode.key.text`, since the key might also be a `tag[myKey]`. But maybe this is fine.

Might be best to use this though https://github.com/getsentry/sentry/blob/a5488bdad9880aebb3cbe19ec0b701945ecdcb13/static/app/components/searchSyntax/parser.tsx#L283-L302

But let's first move it into the searchSyntax/utils

Reply:
sure thing so we wanna move `getKeyName` to `/utils` or were you talking about something else?

Reply:
Yeah, utils in `searchSyntax` component